otx/algorithms/detection/adapters/mmdet/utils/config_utils.py

```diff
@@ -297,47 +229,50 @@ def set_data_classes(config: Config, labels: List[LabelEntity]):
 def patch_datasets(config: Config, domain: Domain):
     """Update dataset configs."""
 
-    def patch_color_conversion(pipeline):
-        # Default data format for OTX is RGB, while mmdet uses BGR, so negate the color conversion flag.
-        for pipeline_step in pipeline:
-            if pipeline_step.type == "Normalize":
-                to_rgb = False
-                if "to_rgb" in pipeline_step:
-                    to_rgb = pipeline_step.to_rgb
-                to_rgb = not bool(to_rgb)
-                pipeline_step.to_rgb = to_rgb
-            elif pipeline_step.type == "MultiScaleFlipAug":
-                patch_color_conversion(pipeline_step.transforms)
-
     assert "data" in config
     for subset in ("train", "val", "test"):
```

Comment:
This implementation assumes that maximum depth of dataset is 2(e.g. Repeat(COCO())).
It is vulnerable to corner case. I think it should be more general
However it is above this PR's range, just keep in mind this...

Reply:
It looks like we need a whole align and refactoring of dataset configuration. Along with datumaro integration, I think we should refer to this part as well.